packages/primitive/address.ts

```diff
@@ -0,0 +1,55 @@
+import { getAddress } from "ethers"
+
+import type { HexString } from "~typing"
+
+interface NodeProvider {
+  getBalance(address: HexString): Promise<bigint>
+  getCode(address: HexString): Promise<HexString>
+}
+
+export type AddressLike = Address | HexString
+
+export class Address {
+  public static wrap(address: AddressLike) {
+    if (address instanceof Address) {
+      return new Address(address.unwrap())
+    }
+    return new Address(getAddress(address))
+  }
+
+  private constructor(private address: HexString) {}
+
+  public ellipsize() {
+    return `${this.address.substring(0, 7)}...${this.address.substring(
+      this.address.length - 5,
+      this.address.length
+    )}`
+  }
+
+  /**
+   * @dev Implement `ethers.Addressable` interface
+   */
+  public async getAddress(): Promise<string> {
+    return this.unwrap()
+  }
```

Comment:
I think maybe we don't need `getAddress()` here, just using `toString()` is fine.

I check the ethers doc for [`interface Addressable`](https://docs.ethers.org/v6/api/address/#Addressable)
> An interface for objects which have an address, and can resolve it asyncronously.
> This allows objects such as [Signer](https://docs.ethers.org/v6/api/providers/#Signer) or [Contract](https://docs.ethers.org/v6/api/contract/#Contract) to be used most places an address can be, for example getting the [balance](https://docs.ethers.org/v6/api/providers/#Provider-getBalance).

My understanding is `ethers.Addressable` interface is for object like Signer or Contract. 
In our case, we will declare in [`AccountFactory`](https://github.com/pilagod/waallet-extension/blob/main/packages/account/factory.ts#L4) or [`Account`](https://github.com/pilagod/waallet-extension/blob/main/packages/account/index.ts#L52)

Reply:
if we remove `getAddress()`, linter will show errors.
ex: [PasskeyAccountFactory ](https://github.com/pilagod/waallet-extension/blob/61e83279f7c8111215fb85eae428497c6d82fce6/packages/account/PasskeyAccount/factory.ts#L28)